mktrayicon.c

```diff
@@ -320,79 +328,92 @@ gpointer watch_fifo(gpointer argv)
 #ifdef DEBUG
 				printf("Removing onmenu handler\n");
 #endif
+				free(param);
 				break;
 			}
 
-			// This block makes sure that the parameter after 'm' is well defined
-			int commas = 0;
-			int bars = 0;
-			if (len < 3){
-				break;
-			}
-			if (param[0] == ',' || param[0] == '|'){
-				printf("Action string cannot begin with ',' or '|'. Use '\\' to escape\n");
-				break;
-			}
-			if (param[len-2] == '|' && param[len-3] != '\\'){
-				printf("'|' can't be second of last. Use '\\' to escape\n");
-				break;
+			// This block makes sure that the parameter after 'm' is ready to be processed
+			
+			char* initial = param; 
+			//we save so that we can free later even if we increase the pointer
+			
+			while (param[0] == ',' || param[0] == '|'){
+				// ignore delimiter characters in first spot
```

Comment:
I'm not sure I follow why this is necessary? If the user puts `,` as the first character, then I think we should just treat that as any other empty label. Not clear that special-casing the _first_ one serves much of a purpose. Maybe _later_ filtering out any menu items with an empty label would be a better way to go?

Reply:
You have a point here. As I explain at the other comment below, I haven't made it to work with empty labels (meaning a blank spot at the menu, with or without action). Do we really need something line that?

But it works well, I think, with labels without action, an info spot at the menu, like you had suggested at the first commit.

Reply:
I think it's okay for us to simply ignore (and maybe warn) if we detect a menu item with an empty label. But I don't think we should do it in the matching up here (and thus only at the start/end). Instead, I think we should parse all the entries, and then warn/ignore on _any_ menu item with an empty label after we've parsed them.